### PR TITLE
fix(elements): Improve OTP a11y

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

- Add `length` prop to OTP inputs to automatically set minLength and maxLength
- Remove unnecessary tabIndex
- Use length for pattern
- Add `useFocus` hook for future usage

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/react/common/form/index.tsx [hunk 1]
```diff
@@ -160,17 +159,23 @@ const useInput = ({ name: inputName, value: initialValue, type: inputType, ...pa
 
   let props = {};
   if (inputType === 'otp') {
+    const p = passthroughProps as Omit<OTPInputProps, 'name' | 'value' | 'type'>;
+    const length = p.length || OTP_LENGTH_DEFAULT;
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/index.tsx:163]: You can use `length` prop to configure the OTP input now

File: packages/elements/src/react/common/form/otp.tsx [hunk 2]
```diff
@@ -72,7 +78,6 @@ const OTPInputSegmented = React.forwardRef<HTMLInputElement, Required<Pick<OTPIn
         <RadixControl
           ref={innerRef}
           {...rest}
-          maxLength={maxLength}
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/otp.tsx:75]: This gets already passed through in the `useInput` hook

File: packages/elements/src/react/hooks/use-focus.hook.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+export const useFocus = (inputRef: React.RefObject<HTMLInputElement>) => {
```
Comment (LekoArts) [inline packages/elements/src/react/hooks/use-focus.hook.ts:3]: For future usage but fitted the a11y theme

**Review by tmilewski — APPROVED** [review]

File: packages/elements/src/react/common/form/otp.tsx [hunk 4]
```diff
@@ -100,8 +105,8 @@ const OTPInputSegmented = React.forwardRef<HTMLInputElement, Required<Pick<OTPIn
             pointerEvents: 'none',
           }}
         >
-          {Array.from({ length: maxLength }).map((_, i) => (
-            <Slot key={i}>
+          {Array.from({ length }).map((_, i) => (
+            <React.Fragment key={i}>
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply packages/elements/src/react/common/form/otp.tsx:109]: 👍 Will do in another PR

**Pull request merged**